tests/logtab.test.js

```diff
@@ -0,0 +1,143 @@
+      procID={undefined}
+      pyModuleName={primaryPythonLogger}
+      terminateInvestProcess={() => {}}
+      sidebarFooterElementId="divID"
+    />
+  );
+  return utils;
+}
+
+describe('LogTab', () => {
+  let workspace;
+  let logfilePath;
+  const logfileName = 'logfile.txt';
+  const uniqueText = 'utils.prepare_workspace';
+  const primaryPythonLogger = 'natcap.invest.hydropower.hydropower_water_yield';
+
+  const logText = `
+2021-01-15 07:14:37,147 (natcap.invest.utils) ${uniqueText}(124) INFO Writing log ...
+2021-01-15 07:14:37,147 (__main__) cli.main(521) Level 100 Starting model with parameters: 
+Arguments for InVEST ${primaryPythonLogger} 3.9.0.post147+gcc5a7cfe:
+biophysical_table_path        C:/Users/dmf/projects/invest/data/biophysical_table_gura.csv
+workspace_dir                 C:/Users/dmf/projects/invest-workbench/runs/awy
+
+2021-01-15 07:14:37,148 (${primaryPythonLogger}) hydropower_water_yield.execute(268) INFO Validating arguments
+2021-01-15 07:14:37,148 (natcap.invest.validation) validation._wrapped_validate_func(915) INFO ...
+2021-01-15 07:14:37,525 (taskgraph.Task) Task.__init__(333) WARNING the ...
+2021-01-15 07:14:37,636 (pygeoprocessing.geoprocessing) geoprocessing.align_and_resize_raster_stack(795) INFO ...
+
+2021-01-19 14:08:32,779 (taskgraph.Task) Task.add_task(781) ERROR Something went wrong when adding task ...
+Traceback (most recent call last):
+  File site-packages/natcap/invest/utils.py, line 860, in reclassify_raster
+  File site-packages/pygeoprocessing/geoprocessing.py, line 1836, in reclassify_raster
+  File site-packages/pygeoprocessing/geoprocessing.py, line 438, in raster_calculator
+  File site-packages/pygeoprocessing/geoprocessing.py, line 1829, in _map_dataset_to_value_op
+pygeoprocessing.geoprocessing.ReclassificationMissingValuesError: (The following 1 raster values [8] from ...
+
+During handling of the above exception, another exception occurred:
+
+Traceback (most recent call last):
+  File site-packages/taskgraph/Task.py, line 747, in add_task
+  File site-packages/taskgraph/Task.py, line 1234, in _call
+  File site-packages/natcap/invest/sdr/sdr.py, line 997, in _calculate_w
+  File site-packages/natcap/invest/utils.py, line 868, in reclassify_raster
+ValueError: Values in the LULC raster were found that are not represented under the lucode column of the Biophysical table....
+2021-01-19 14:08:32,780 (natcap.invest.utils) utils.prepare_workspace(130) INFO Elapsed time: 3.5s
+`;
+
+  beforeEach(() => {
+    workspace = fs.mkdtempSync(path.join('tests/data', 'log-'));
+    logfilePath = path.join(workspace, logfileName);
+    fs.writeFileSync(logfilePath, logText);
+  });
+
+  afterEach(() => {
+    cleanupDir(workspace);
+  });
+
+  test('Text in logfile is rendered', async () => {
+    const { findByText } = renderLogTab(
+      logfilePath, primaryPythonLogger
+    );
+
+    const log = await findByText(new RegExp(uniqueText));
+    expect(log).toBeInTheDocument();
```

Comment:
Hey Dave, what's this do `toBeInTheDocument()`?

Reply:
Specifically, what's the document?

Reply:
@dcdenu4 That'd be the "D" in DOM. i.e. the html page that is rendered by react. These `findBy` queries are querying the DOM and returning a specific node. Then, we're just expecting the node `toBeInTheDocument`. In this case, `findByText` would actually have raised an error if it couldn't find the text, so the `expect` is a little extraneous. But it would be weird to leave it out, I think.

The queries are from `testing-library` 
https://testing-library.com/docs/queries/about/

The "matchers" (`expect().toBe...`) are from jest or jest-dom
https://jestjs.io/docs/en/expect
https://github.com/testing-library/jest-dom